src/zns.ts

```diff
@@ -69,12 +69,23 @@ export default class {
     );
 
     if (!registryRecord) return null;
-    const [ownerAddress, resolverAddress] = registryRecord.arguments as [
+    let [ownerAddress, resolverAddress] = registryRecord.arguments as [
       string,
       string
     ];
     const resolution = await this.getResolverRecordsStructure(resolverAddress);
     const addresses = _.mapValues(resolution.crypto, 'address');
+    // at the moment ownerAddress is publicKey which starts with 0x 
+    if (ownerAddress.startsWith('0x')) {
+
+      // If it is uncompressed i have to compress it 
+      if (/^(0x)?(04)?[a-f0-9]{128}$/i.test(ownerAddress)) {
```

Comment:
As owner is a zil address, it can not be compressed: it has a constant length of 40 + 2

Reply:
Does it mean I should make sure the length is 40 + 2 or this .startsWith('0x') check is enough?